lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java

```diff
@@ -951,6 +979,29 @@ private TaskImpl checkConcurrencyAndSetModified(TaskImpl newTaskImpl, TaskImpl o
     return newTaskImpl;
   }
 
+  private TaskImpl terminateCancelCommonActions(String taskId, TaskState targetState)
+      throws NotAuthorizedException, TaskNotFoundException, InvalidStateException {
+    if (taskId == null || taskId.isEmpty()) {
+      throw new TaskNotFoundException(taskId, String.format(TASK_WITH_ID_WAS_NOT_FOUND, taskId));
+    }
+    TaskImpl task = (TaskImpl) getTask(taskId);
+    TaskState state = task.getState();
+    if (state.isEndState()) {
+      throw new InvalidStateException(String.format(TASK_WITH_ID_IS_ALREADY_IN_END_STATE, taskId));
+    }
+
+    Instant now = Instant.now();
+    task.setOwner(null);
+    task.setModified(now);
+    task.setCompleted(now);
+    task.setClaimed(null);
+    task.setRead(true);
```

Comment:
Why this. We don't know if the task has been read, do we?

Reply:
ok, retained also the read timestamp